core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -0,0 +1,233 @@
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<Block> iterator() {
+    RocksIterator iter =
+        mDb.newIterator(mBlockMetaColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seekToFirst();
+    return new Iterator<Block>() {
+      @Override
+      public boolean hasNext() {
+        return iter.isValid();
+      }
+
+      @Override
+      public Block next() {
+        try {
+          return new Block(Longs.fromByteArray(iter.key()), BlockMeta.parseFrom(iter.value()));
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        } finally {
+          iter.next();
+        }
+      }
+    };
+  }
+
+  // Implementation based on
+  // https://google.github.io/guava/releases/21.0/api/docs/src-html/com/google/common/primitives/
+  // Longs.html#line.267
+  private byte[] toByteArray(long long1, long long2) {
+    byte[] key = new byte[16];
+    for (int i = 7; i >= 0; i--) {
+      key[i] = (byte) (long1 & 0xffL);
+      long1 >>= 8;
+    }
+    for (int i = 15; i >= 8; i--) {
+      key[i] = (byte) (long2 & 0xffL);
+      long2 >>= 8;
+    }
+    return key;
+  }
+
+  private void initDb(String baseDir) throws RocksDBException {
+    if (mDb != null) {
+      // Column handles must be closed before closing the db, or an exception gets thrown.
+      mDefaultColumn.close();
+      mBlockMetaColumn.close();
+      mBlockLocationsColumn.close();
+      mDb.close();
+      try {
+        FileUtils.deletePathRecursively(mDbPath);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    new File(baseDir).mkdirs();
+
+    ColumnFamilyOptions cfOpts = new ColumnFamilyOptions()
+        .useFixedLengthPrefixExtractor(8); // We always search using the initial long key
+
+    List<ColumnFamilyDescriptor> cfDescriptors = Arrays.asList(
+        new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY, cfOpts),
+        new ColumnFamilyDescriptor("block-meta".getBytes(), cfOpts),
+        new ColumnFamilyDescriptor("block-locations".getBytes(), cfOpts)
+    );
+
+    DBOptions options = new DBOptions()
+        .setCreateIfMissing(true)
+        .setCreateMissingColumnFamilies(true);
+
+    // a list which will hold the handles for the column families once the db is opened
+    List<ColumnFamilyHandle> columns = new ArrayList<>();
+    mDbPath = newDbPath(baseDir);
+    mDb = RocksDB.open(options, mDbPath, cfDescriptors, columns);
+    mDefaultColumn = columns.get(0);
+    mBlockMetaColumn = columns.get(1);
+    mBlockLocationsColumn = columns.get(2);
+
+    LOG.info("Created new rocks database under path {}", mDbPath);
+  }
+
+  private static String newDbPath(String baseDir) {
+    // We use a unique subdirectory to avoid conflicts with the previous RocksDB instance when we
+    // re-create the db in initDB().
+    return PathUtils.concatPath(baseDir,
+        "blocks-" + System.nanoTime() + "-" + CommonUtils.randomAlphaNumString(3));
```

Comment:
better to be in a util method, so file and other stores can leverage the same string format

Reply:
Will refactor once we have a second store.